src/Hazelcast.Net/Serialization/Compact/RabinFingerprint.cs

```diff
@@ -66,10 +72,13 @@ public static ulong Fingerprint(ulong fingerprint, int value)
         // fingerprints a string value
         public static ulong Fingerprint(ulong fingerprint, string value)
         {
-            if (value == null) return Fingerprint(fingerprint, 0);
+            if (value == null) throw new ArgumentNullException(nameof(value));
```

Comment:
This class ~is~ should be marked with a `#nullable enable` line, which is missing and I am going to add it, and then it means that a `string` parameter cannot be `null` (it would have to be `string?`) -- however, as a general rule of best-practice, we tend to keep all null checks in .NET because these "non-null" things are only compile-time verifications, and there are ways that at runtime shit happens. I know (discussed with Asim) that in many places in the Java code, tests have been removed because "things will be all right" -- we tend to keep them everywhere in .NET.